### DISCUSSION TITLE
Easiest way to render app to html (build a single html page from components)

### DISCUSSION THREAD

**Discussion by codenomnom** [vuejs/docs#1664] [General]
Hi guys,
I know there are tons of static page generators out there. But my goal is to use Vue to render the entire tree of components and convert that to a simple HTML result that I can later put inside an HTML file. So it would be a single static page generation :) Something similar to 11ty, I'd guess🤔 

Any ideas on what would be the easiest way for that?

**Comment by leopiccionia** [comment]
Have you tried using the [`renderToString`](https://vuejs.org/api/ssr.html#rendertostring) function?
